WalletWasabi.Fluent/Models/UI/ApplicationSettings.cs

```diff
@@ -239,8 +235,12 @@ private void Save()
 			{
 				try
 				{
-					PersistentConfig currentConfig = PersistentConfigManager.LoadFile(_persistentConfigFilePath);
-					PersistentConfig newConfig = ApplyChanges(currentConfig);
+					var loadedConfig = PersistentConfigManager.LoadFile(_persistentConfigFilePath);
+					if (loadedConfig is not PersistentConfig currentConfig)
+					{
+						throw new NotSupportedException("Only configuration files after v2.5.1 are supported.");
```

Comment:
when I downgrade (switch to current master after running the PR) I get that the "wallet is corrupted".
It does not throw exception related to config file

of course downgrading is not a big worry, just commenting here

Reply:
Just "Wallet is corrupted" without any other hint? What changed in your wallet files?

Reply:
No, it is not okay but it is not because this PR, it is just that you found it while testing this PR.